proposals/config_ownership_crd.md

```diff
@@ -0,0 +1,129 @@
+# CRD for configuration ownership in brownfield deployments
+##### Deploying Ingress Controller to hybrid environments with partial Application Gateway configuration ownership.
+
+
+### Document Purpose
+This document is a proposal for the creation of 2 new [CRDs](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) with the goal of
+augmenting the configuration of [Azure Application Gateway Ingress Controller](https://azure.github.io/application-gateway-kubernetes-ingress/).
+
+  - Author: [Delyan Raychev](https://github.com/draychev)
+  - Published: June 2nd, 2019
+  - Status:  Open for comments
+
+### Problem Statement
+As of this writing Ingress Controller (v0.6.0) blindly overwrites existing Application Gateway configurations when applying changes to:
+  - [listeners](https://github.com/Azure/application-gateway-kubernetes-ingress/blob/9820ff8aed758b2626c068e2fb25629e06540159/pkg/controller/controller.go#L100)
+  - [routing rules](https://github.com/Azure/application-gateway-kubernetes-ingress/blob/9820ff8aed758b2626c068e2fb25629e06540159/pkg/controller/controller.go#L107)
+  - [HTTP settings](https://github.com/Azure/application-gateway-kubernetes-ingress/blob/9820ff8aed758b2626c068e2fb25629e06540159/pkg/controller/controller.go#L83)
+  - [backend pools](https://github.com/Azure/application-gateway-kubernetes-ingress/blob/9820ff8aed758b2626c068e2fb25629e06540159/pkg/controller/controller.go#L90)
+  - [health probes](https://github.com/Azure/application-gateway-kubernetes-ingress/blob/9820ff8aed758b2626c068e2fb25629e06540159/pkg/controller/controller.go#L76)
+
+Ingress Controller assumes full ownership and control of the given App Gateway. This makes it impossible to deploy it
+to environments with existing pre-configured App Gateways (brownfield deployments). It is also not possible to use AGIC
+in hybrid environments where Application Gateway configuration is managed in parts by system administrators as well as
+Ingress Controller.
+
+### Proposed Solution
+We propose the creation of two new Kubernetes custom resource definitions ([CRD](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/)):
+  - `AzureIngressManagedListener` - defines a reference to a single listener. Ingress Controller will assume ownership
```

Comment:
Would it be better to have all allowed `targets/rules/locations` in one place ?

Reply:
Talked offline - decided to keep one CRD per ip/port/host/[urls]